File: client/task-master/src/app/models/Board.ts
import { CardList } from './CardList';
import { User } from './User';
import { DesignType } from './enum/DesignType';

export class Board {
  // Идентификатор доски
  id: string;
  // Название доски
  title?: string;
  // Дата создания доски
  createdAt?: Date;
  // Дата последнего обновления доски
  lastUpdated?: Date;
  // Код цвета доски
  colorCode?: string;
  // URL изображения доски
  imageUrl?: string;
  // Тип дизайна доски
  designType: DesignType;
  // Владелец доски
  owner?: User;
  // Списки карточек на доске
  cardLists: CardList[] = [];

  // Конструктор объекта доски
  constructor(board?: Board) {
    // Если передана исходная доска
    if (board != undefined) {
      // Присвоение значений из исходной доски
      this.id = board?.id;
      this.title = board?.title;
      this.createdAt = board?.createdAt;
      this.lastUpdated = board?.lastUpdated;
      this.colorCode = board?.colorCode;
      this.designType = board?.designType;
      this.owner = board?.owner;
      this.cardLists = board?.cardLists ?? this.cardLists;
    }
  }

  // Метод для присвоения значений из частичной доски
  assign(board: Partial<Board>) {
    this.id = board?.id ?? this.id;
    this.title = board?.title ?? this.title;
    this.createdAt = board?.createdAt ?? this.createdAt;
    this.lastUpdated = board?.lastUpdated ?? this.lastUpdated;
    this.colorCode = board?.colorCode ?? this.colorCode;
    this.imageUrl = board?.imageUrl ?? this.imageUrl;
    this.designType = board?.designType ?? this.designType;
    this.owner = board?.owner ?? this.owner;
    this.cardLists = board?.cardLists ?? this.cardLists;
  }
}

File: server/TaskMaster/TaskMaster.DataWebApi/Models/Board.cs
using TaskMaster.DataAccessModule.Constants;

namespace TaskMaster.DataWebApi.Models
{
	/// <summary>
	/// Модель доски.
	/// </summary>
	public class Board
	{
		/// <summary>
		/// Идентификатор доски.
		/// </summary>
		public Guid Id { get; set; }

		/// <summary>
		/// Заголовок доски.
		/// </summary>
		public string? Title { get; set; }

		/// <summary>
		/// Дата создания доски.
		/// </summary>
		public DateTime? CreatedAt { get; set; }

		/// <summary>
		/// Дата последнего обновления доски.
		/// </summary>
		public DateTime? LastUpdated { get; set; }

		/// <summary>
		/// Код цвета доски.
		/// </summary>
		public string? ColorCode { get; set; }

		/// <summary>
		/// URL изображения доски.
		/// </summary>
		public string? ImageUrl { get; set; }

		/// <summary>
		/// Тип дизайна доски.
		/// </summary>
		public DesignType DesignType { get; set; }

		/// <summary>
		/// Владелец доски.
		/// </summary>
		public User Owner { get; set; }
	}
}
